packages/core/utils/src/dml/helpers/entity-builder/define-relationship.ts

```diff
@@ -492,22 +475,36 @@ export function defineManyToManyRelationship(
         .join("_")
   }
 
+  let isOwner: boolean | undefined = undefined
+
+  /**
+   * If any of the following properties are provided, we consider
+   * the current side to be the owner
+   */
+  if (joinColumn || inverseJoinColumn || relationship.options.pivotTable) {
+    isOwner = true
+  }
+
+  /**
+   * If any of the properties are provided on the other side,
+   * then we do not expect the current side to be the owner
+   */
   const otherSideRelationOptions = otherSideRelationship.parse("").options
+  if (
+    isOwner === undefined &&
+    (otherSideRelationOptions.pivotTable ||
+      otherSideRelationOptions.joinColumn ||
+      otherSideRelationOptions.inverseJoinColumn)
+  ) {
+    isOwner = false
+  }
```

Comment:
Suggestion: We could probably check if the configuration is on both side and in that case compose an error message to explain that it should only be configured on one side (the owner side)